bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Tree.java

```diff
@@ -757,24 +757,24 @@ LRESULT CDDS_ITEMPOSTPAINT (NMTVCUSTOMDRAW nmcd, long wParam, long lParam) {
 						Image [] images  = item.images;
 						if (images != null) image = images [index];
 					}
-					int inset = i != 0 ? INSET : 0;
-					int offset = i != 0 ? INSET : INSET + 2;
+					int inset = i != 0 ? DPIUtil.autoScaleUp(INSET, getZoom()) : 0;
+					int offset = i != 0 ? DPIUtil.autoScaleUp(INSET, getZoom()) : DPIUtil.autoScaleUp(INSET + 2, getZoom());
 					if (image != null) {
 						Rectangle bounds = image.getBounds (); // Points
-						if (size == null) size = DPIUtil.autoScaleDown (getImageSize ()); // To Points
+						if (size == null) size = DPIUtil.scaleDown (getImageSize (), getZoom()); // To Points
 						if (!ignoreDrawForeground) {
 							//int y1 = rect.top + (index == 0 ? (getItemHeight () - size.y) / 2 : 0);
-							int y1 = rect.top + DPIUtil.autoScaleUp((getItemHeight () - size.y) / 2);
+							int y1 = rect.top + DPIUtil.autoScaleUp((getItemHeight () - size.y) / 2, getZoom());
 							int x1 = Math.max (rect.left, rect.left - inset + 1);
 							GCData data = new GCData();
 							data.device = display;
-							GC gc = getNewGC(hDC, data);
-							gc.setClipping (DPIUtil.autoScaleDown(new Rectangle(x1, rect.top, rect.right - x1, rect.bottom - rect.top)));
-							gc.drawImage (image, 0, 0, bounds.width, bounds.height, DPIUtil.autoScaleDown(x1), DPIUtil.autoScaleDown(y1), size.x, size.y);
+							GC gc = GC.win32_new (hDC, data);
```

Comment:
This seems to revert changes of a previous commit. Is that intended?

Reply:
Nope. Probably because of rebasing. Removed it.